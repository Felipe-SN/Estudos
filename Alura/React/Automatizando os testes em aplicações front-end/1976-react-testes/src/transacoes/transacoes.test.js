import React from 'react';
import { render } from '@testing-library/react';
import Transacoes from './Transacoes';

describe('Componente de lista de transações do extrato', () => {
  it('O snapshot do componente deve permanecer sempre o mesmo', () => {
    const transacoes = [
      {
        data: '10/06/2022',
        id: 1,
        transacao: 'saque',
        valor: '10',
      },
      {
        data: '26/07/2022',
        id: 2,
        transacao: 'deposito',
        valor: '20',
      },
    ];

    const { container } = render(<Transacoes transacoes={transacoes} />);

    expect(container.childNodes).toMatchSnapshot();
  });
});
