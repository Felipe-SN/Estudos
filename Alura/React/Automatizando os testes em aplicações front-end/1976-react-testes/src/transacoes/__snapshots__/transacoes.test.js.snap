// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de lista de transações do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
NodeList [
  <div
    data-testid="transacoes"
  >
    <div
      class="transacao-container"
    >
      <p>
        10/06/2022
      </p>
      <p>
        saque
      </p>
      <p>
        R$ 
        10
      </p>
    </div>
    <div
      class="transacao-container"
    >
      <p>
        26/07/2022
      </p>
      <p>
        deposito
      </p>
      <p>
        R$ 
        20
      </p>
    </div>
  </div>,
]
`;
